src/t_zset.c

```diff
@@ -1232,6 +1232,22 @@ unsigned long zsetLength(const robj *zobj) {
     return length;
 }
 
+size_t zsetAllocSize(const robj *o) {
+    debugServerAssertWithInfo(NULL,o,o->type == OBJ_ZSET);
```

Comment:
i think we shouldn't use debug assert here.
Only when the code has a significant impact on the production environment will we use debug assert; otherwise, it should be redisAssert.